kite-morphlines/kite-morphlines-maxmind/src/main/java/org/kitesdk/morphline/maxmind/GeoIPBuilder.java

```diff
@@ -99,18 +101,23 @@ protected boolean doProcess(Record record) {
         } catch (IOException e) {
           throw new MorphlineRuntimeException("Cannot perform GeoIP lookup for IP: " + addr, e);
         }
-        
-        ObjectNode location = (ObjectNode) json.get("location");
-        if (location != null) {
-          JsonNode jlatitude = location.get("latitude");
-          JsonNode jlongitude = location.get("longitude");
-          if (jlatitude != null && jlongitude != null) {
-            String latitude = jlatitude.toString();
-            String longitude = jlongitude.toString();
-            location.put("latitude_longitude", latitude + "," + longitude);
-            location.put("longitude_latitude", longitude + "," + latitude);
+
+        if (json != null) {
+          ObjectNode location = (ObjectNode) json.get("location");
+          if (location != null) {
+            JsonNode jlatitude = location.get("latitude");
+            JsonNode jlongitude = location.get("longitude");
+            if (jlatitude != null && jlongitude != null) {
+              String latitude = jlatitude.toString();
+              String longitude = jlongitude.toString();
+              location.put("latitude_longitude", latitude + "," + longitude);
+              location.put("longitude_latitude", longitude + "," + latitude);
+            }
           }
-        }        
+        } else {
+          json = objectMapper.createObjectNode();
```

Comment:
Seems like doProcess() should do "return false" rather than continue with an empty json object. This way an enclosing tryRules command (or "if" command) could react and handle the case whichever way desired. This is similar to how, say, the extractURIParameters command handles invalid data. Thoughts?